design/design.md

```diff
@@ -0,0 +1,175 @@
+>* `ShipToken`
+
+### PelletState UTxO:
+
+>#### Address
+>* Parameterized on the admin token.
+>
+>#### Datum
+>* pos_x: **Int**
+>* pos_y: **Int**
+>* fuel: **Int**
+>* ship_token_policy: **PolicyId**
+>
+>#### Value
+>* minAda
+>* admin token
+
+### Pot UTxO:
+
+>#### Address
+>* Parameterized on the admin token.
+>
+>#### Datum
+>* ship_token_policy: **PolicyId**
+>
+>#### Value
+>* minAda
+>* reward amount.
+>* admin token
+>
+>Note: the reward amount is updated during the course of the game, whenever a new user joins or reaches the pot.
+
+
+## Transactions
+
+### Create Pot UTxO:
+This transaction creates the unique `Pot` UTxO locking min ada and an admin token.
+
+![createPot diagram](img/createPot.png)
+
+
+### Create a PelletState UTxO:
+Creates one `PelletState` UTxO locking min ada and an admin token, setting in the datum the `pos_x` and `pos_y` coordinates where the pellet will be located on the grid and the `fuel` value equal to some initial value.
+
+![createPellet diagram](img/createPellet.png)
+
+
+### Create a ShipState UTxO:
+Creates a `ShipState` UTxO locking min ada and a ship token (minted in this tx), specifying in the datum the initial `pos_x` and `pos_y` coordinates of the ship, and setting `fuel` to an initial amount. Also adds to the `Pot` UTxO value the inscription fee paid by the user.
+
+![createShip diagram](img/createShip.png)
+
+
+### Move a Ship:
+Updates the `pos_x`, `pos_y` and `fuel` datum fields of the `ShipState` UTxO by adding the `delta_x` and `delta_y` values specified in the redeemer, and subtracting the fuel amount needed for the displacement.
+
+![moveShip diagram](img/moveShip.png)
+
+
+### Gather Fuel:
+Updates the `fuel` datum field of both the `ShipState` and `PelletState` UTxOs, adding the `amount` (specified in the redeemer) from the first and subtracting it from the latter.
+
+![gatherFuel diagram](img/gatherFuel.png)
+
+
+### Collect Rewards:
+Subtracts from the `Pot` UTxO 50% of the ada value, and pays that amount to the owner of the ship that reached the pot, together with the min ada locked in the `ShipState` UTxO. The ship token is burnt.
+
+![collect diagram](img/collect.png)
+
+
+### Quit Game:
+Pays the min ada locked in the `ShipState` UTxO back to the ship owner and burns the ship token.
+
+![quit diagram](img/quit.png)
+
+
+## Validators & Minting Policies
+
+### Pot validator:
+* Params: admin token.
+
+#### *AddNewPlayer Redeemer*
+* `Pot` output value equals input value plus the inscription fee.
+* adminToken is in the input.
+* datum doesn't change.
+
+#### *Collect Redeemer (includes address of the player that reached the pot)*
+* ship token is present in some input.
+* `Pot` output value has at most 50% adas less than input value.
+
+### PelletState validator:
+* Params: admin token.
+
+#### *Gather Redeemer (includes gathering amount)*
+* ship token is present in some input.
+* there is a `ShipState` input with the same x and y datum coordinates as the `PelletState` UTxO.
```

Comment:
This should be checked in the ship validator, not here. The rule should be "ship validator reads the pellet datum, but not the other way"